snuba/datasets/storages/querylog.py

```diff
@@ -50,6 +50,12 @@
                     # ``Cache.get_readthrough`` query execution path. See GH-902.
                     ("is_duplicate", UInt(8)),
                     ("consistent", UInt(8)),
+                    ("all_columns", Array(LowCardinality(String()))),
```

Comment:
Should these types line up with the types in the migration? Shouldn't this be `Array(Array(LowCardinality(String())))`

Reply:
no, they mean a different thing.
On table creation clickhouse allows for nested columns. A nested column represent a table in a table. Here the nested column is `clickhouse_queries` that has all these subfields. So `clickhouse_queries` if implicitly an array of clickhosue queries, each of which has all these columns. `all_columns` is an array of columns. And we want an array of column per clickhouse query.

Now the nested column idea is mostly syntactic sugar in the clickhouse create table statement to identify a set of arrays, one per column. 
if you run a `describe querylog_local` statement, you will so no trace of the `clickhouse-queries` nested column. You will see `clickhouse.queries.sql Array(String)`, `clickhouse_queries.all_columns Array(Array(LowCardinality(String))` etc. instead.

Thus when defining a schema for creation from scratch (here), we can use the nested construct. While when we add columns we need to deal with individual columns, and add `clickhouse_queries.all_columns` as array of arrays. 

Some details are here: https://clickhouse.tech/docs/en/sql-reference/data-types/nested-data-structures/nested/